smithy-build/src/main/java/software/amazon/smithy/build/transforms/FlattenNamespaces.java

```diff
@@ -158,9 +158,13 @@ private ShapeId updateNamespace(ShapeId shapeId, String namespace) {
     }
 
     private Map<ShapeId, ShapeId> getRenamedShapesConnectedToService(Config config, Model model) {
-        Walker shapeWalker = new Walker(model.getKnowledge(NeighborProviderIndex.class, NeighborProviderIndex::new)
-                .getProvider());
-        ServiceShape service = model.expectShape(config.getService(), ServiceShape.class);
+        Walker shapeWalker = new Walker(NeighborProviderIndex.of(model).getProvider());
+        Shape service = model.expectShape(config.getService());
```

Comment:
This could be `model.expectShape(config.getService(), ServiceShape.class())`

Reply:
Fixed.